fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -292,3 +299,112 @@ impl IBitcoindRpc for ElectrumClient {
         })
     }
 }
+
+// TODO: (@oleonardolima) Should both this EsploraClient and ElectrumClient be extracted to different files/namespaces ?
+pub struct EsploraClient(esplora_client::AsyncClient);
+
+impl EsploraClient {
+    fn new(url: &Url) -> anyhow::Result<Self> {
+        let builder = esplora_client::Builder::new(url.as_str());
+        let client = builder.build_async()?;
+        Ok(Self(client))
+    }
+}
+
+impl fmt::Debug for EsploraClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("EsploraClient")
```

Comment:
Would be interesting to see the API url if possible

Reply:
Great point, I'll try to check if that's possible 😁

Reply:
Bump

Reply:
@elsirion I've tried to work on this one, it's possible to get the underlying URL, e.g https://blockstream.info/testnet/api as it can be accessed by the `self.0.url()` from `AsyncClient`.

But having access to each different path it's not possible, as it's only accessible from the [`Request` struct in `reqwest`](https://docs.rs/reqwest/latest/src/reqwest/async_impl/request.rs.html#526-549)

https://github.com/seanmonstar/reqwest/blob/bdd4db02447c68b6be528c25cde6fa1fc47bf8e1/src/async_impl/request.rs#L526-L549

Do you think that adding only the underlying API url would be enough?

Reply:
I don't get it.

Does this work:

```
pub struct EsploraClient {
  client: esplora_client::AsyncClient,
  sanitized_url: String,
}
```

?

Reply:
For `bitcoind`, I used:

```
 let retry_client = RetryClient::new(
        Client(ErrorReporting::new(url, bitcoind_client)),
        task_handle,
    );
```

to wrap the original client into something that reports error. You could do the similiar thing here.

Reply:
I mean this is possible:
``` rust
impl fmt::Debug for EsploraClient {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.write_fmt(format_args!("EsploraClient [url: {}]", self.0.url()))
    }
}
```

But it will only show the URL (eg. https://blockstream.info/testnet/api) and not the path being called.